pkg/sharing/sharing.go

```diff
@@ -152,16 +152,32 @@ func (s *Sharing) BeOwner(inst *instance.Instance, slug string) error {
 // CreatePreviewPermissions creates the permissions doc for previewing this sharing,
 // or updates it with the new codes if the document already exists
 func (s *Sharing) CreatePreviewPermissions(inst *instance.Instance) (map[string]string, error) {
+	doc, err := permissions.GetForSharePreview(inst, s.SID)
```

Comment:
[nitpick] You are doing the comparison two times below to know if there was an existing permissions document, and it is not done the same:

```go
if doc != nil {
```

```go
if err == nil {
```

You should stick to one way, let's say `if doc != nil {` and drop `err` declaration here.